execCommand("outdent") on a list containing a table duplicates the content

Ahmad-S792 opened the pull request:
#### 376c0503ca4634a5c9cf39447bb1b8da599a800a
<pre>
execCommand(&quot;outdent&quot;) on a list containing a table duplicates the content

<a href="https://bugs.webkit.org/show_bug.cgi?id=24249">https://bugs.webkit.org/show_bug.cgi?id=24249</a>
<a href="https://rdar.apple.com/problem/129093516">rdar://problem/129093516</a>

Reviewed by NOBODY (OOPS!).

Merge: <a href="https://chromium.googlesource.com/chromium/blink/+/0cd4b3864db4fc9c0abfeeee0775c5fcbbca0184">https://chromium.googlesource.com/chromium/blink/+/0cd4b3864db4fc9c0abfeeee0775c5fcbbca0184</a>

This patch is for cases of indent/outdent for &apos;li&apos; elements containing children
spanning across multiple lines (paragraphs).
For such cases the selection only encompasses children contained within the
first paragraph (within the li), thereby causing incorrect behavior when indenting.
Thus we need to ensure while inserting a new li element and cloning the original,
that all the children are considered.

With this change, an existing test: indent-pre-list.html fails because of the
previous incorrect behavior. Two &apos;li`s&apos; were created for the indented &apos;li&apos;
with the text contained within the first paragraph forming the first and the
remaining content moving to the second &apos;li&apos; element.

This has now been fixed resulting in only a single indented &apos;li&apos;.

* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::tryIndentingAsListItem):
* LayoutTests/editing/execCommand/indent-pre-list-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/run/indent-expected.txt:
* LayoutTests/editing/execCommand/indent-list-item-with-children.html: Add Test Case
* LayoutTests/editing/execCommand/indent-list-item-with-children-expected.txt: Add Test Case Expectation
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/376c0503ca4634a5c9cf39447bb1b8da599a800a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/178553 "Passed style check") | [❌ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/52491 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/188169 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/141802 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/36e92bfa-c64c-4628-a0e0-0050330866cb) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/180416 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/53785 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/52201 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/188169 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/141802 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/34545a6b-744a-47dc-aff2-6aa15cc79223) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/181510 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/155/builds/53785 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/188169 "Hash 376c0503 for PR 30111 does not build (failure)") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bf94a0f2-183e-440f-a6dc-175a12ccb1af) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/155/builds/53785 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/156/builds/52201 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 jsc-wpe](https://ews-build.webkit.org/#/builders/5/builds/188169 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| [❌ 🧪 jsc-x86-64](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/155/builds/53785 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/36624 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| | [❌ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/45091 "Failed to build and analyze WebKit") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/156/builds/52201 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/190596 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| | [❌ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/52160 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/2/builds/190596 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/52841 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/2/builds/190596 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| | [❌ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/160/builds/52841 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/125108 "Failed to build and analyze WebKit") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/119744 "Hash 376c0503 for PR 30111 does not build (failure)") | | 
| | [❌ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/51359 "Hash 376c0503 for PR 30111 does not build (failure)") | [❌ 🧪 mac-site-isolation](https://ews-build.webkit.org/#/builders/168/builds/46286 "Hash 376c0503 for PR 30111 does not build (failure)") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/50744 "Hash 376c0503 for PR 30111 does not build (failure)") | | | | 
| | [❌ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/50984 "Hash 376c0503 for PR 30111 does not build (failure)") | | | | 
| | [❌ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/50806 "Hash 376c0503 for PR 30111 does not build (failure)") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 committed:
execCommand("outdent") on a list containing a table duplicates the content

https://bugs.webkit.org/show_bug.cgi?id=24249
rdar://problem/129093516

Reviewed by NOBODY (OOPS!).

Merge: https://chromium.googlesource.com/chromium/blink/+/0cd4b3864db4fc9c0abfeeee0775c5fcbbca0184

This patch is for cases of indent/outdent for 'li' elements containing children
spanning across multiple lines (paragraphs).
For such cases the selection only encompasses children contained within the
first paragraph (within the li), thereby causing incorrect behavior when indenting.
Thus we need to ensure while inserting a new li element and cloning the original,
that all the children are considered.

With this change, an existing test: indent-pre-list.html fails because of the
previous incorrect behavior. Two 'li`s' were created for the indented 'li'
with the text contained within the first paragraph forming the first and the
remaining content moving to the second 'li' element.

This has now been fixed resulting in only a single indented 'li'.

* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::tryIndentingAsListItem):
* LayoutTests/editing/execCommand/indent-pre-list-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/run/indent-expected.txt:
* LayoutTests/editing/execCommand/indent-list-item-with-children.html: Add Test Case
* LayoutTests/editing/execCommand/indent-list-item-with-children-expected.txt: Add Test Case Expectation

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1870b96c36f08cd1aadb6699b77098acfc179b10. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30111

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5c426230491de8388f8c119aeec227524226b42b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/30111

Ahmad-S792 committed:
execCommand("outdent") on a list containing a table duplicates the content

https://bugs.webkit.org/show_bug.cgi?id=24249
rdar://problem/129093516

Reviewed by NOBODY (OOPS!).

Merge: https://chromium.googlesource.com/chromium/blink/+/0cd4b3864db4fc9c0abfeeee0775c5fcbbca0184

This patch is for cases of indent/outdent for 'li' elements containing children
spanning across multiple lines (paragraphs).
For such cases the selection only encompasses children contained within the
first paragraph (within the li), thereby causing incorrect behavior when indenting.
Thus we need to ensure while inserting a new li element and cloning the original,
that all the children are considered.

With this change, an existing test: indent-pre-list.html fails because of the
previous incorrect behavior. Two 'li`s' were created for the indented 'li'
with the text contained within the first paragraph forming the first and the
remaining content moving to the second 'li' element.

This has now been fixed resulting in only a single indented 'li'.

* Source/WebCore/editing/IndentOutdentCommand.cpp:
(WebCore::IndentOutdentCommand::tryIndentingAsListItem):
* LayoutTests/editing/execCommand/indent-pre-list-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/run/indent-expected.txt:
* LayoutTests/editing/execCommand/indent-list-item-with-children.html: Add Test Case
* LayoutTests/editing/execCommand/indent-list-item-with-children-expected.txt: Add Test Case Expectation
